### PR TITLE
Update to new Spack CachedCMakePackage

### PR DESCRIPTION
**Pull request by white238**
Updates us to the new `CachedCmakePackage` in Spack.  Also updates our axom spack package to the new one.

### CODE REVIEW

**Comment by joshessman-llnl** [comment]
For reference: https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/cached_cmake.py

**Comment by jamiebramwell** [comment]
I'll test this on my WSL setup.

**Comment by jamiebramwell** [comment]
Works on WSL!

File: host-configs/docker/docker-linux-ubuntu16.04-x86_64-gcc@8.1.0.cmake [hunk 1]
```diff
@@ -1,73 +1,100 @@
...
-set(NETCDF_DIR "${TPL_ROOT}/netcdf-c-4.7.4" CACHE PATH "")
-
 set(PARMETIS_DIR "${TPL_ROOT}/parmetis-4.0.3" CACHE PATH "")
 
+set(NETCDF_DIR "${TPL_ROOT}/netcdf-c-4.7.4" CACHE PATH "")
+
 set(SUPERLUDIST_DIR "${TPL_ROOT}/superlu-dist-6.1.1" CACHE PATH "")
 
-set(MFEM_DIR "${TPL_ROOT}/mfem-4.2.0" CACHE PATH "")
+# PETSC not build
```
Comment (jamiebramwell) [inline host-configs/docker/docker-linux-ubuntu16.04-x86_64-gcc@8.1.0.cmake]: I assume this is an upstream spack thing, but `build` -> `built`

File: scripts/spack/packages/serac/package.py [hunk 2]
```diff
@@ -164,327 +137,204 @@ class Serac(CMakePackage, CudaPackage):
...
-            cfg.write(cmake_cache_option("ENABLE_CUDA", True))
-
-            cudatoolkitdir = spec['cuda'].prefix
-            cfg.write(cmake_cache_entry("CUDA_TOOLKIT_ROOT_DIR",
-                                        cudatoolkitdir))
-            cudacompiler = "${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc"
-            cfg.write(cmake_cache_entry("CMAKE_CUDA_COMPILER",
-                                        cudacompiler))
+
+    # def initconfig_compiler_entries(self):
```
Comment (jamiebramwell) [inline scripts/spack/packages/serac/package.py]: Dead code

**Review by jamiebramwell — APPROVED** [review]
Awesome stuff!

File: host-configs/docker/docker-linux-ubuntu16.04-x86_64-gcc@8.1.0.cmake (hunk 1 above)
Comment (white238) [reply host-configs/docker/docker-linux-ubuntu16.04-x86_64-gcc@8.1.0.cmake]: Ack i thought i fixed that.  I'll change it then not rebuild the host-configs

File: scripts/spack/packages/serac/package.py (hunk 2 above)
Comment (white238) [reply scripts/spack/packages/serac/package.py]: I was torn on this one.  My hatred for dead code against if we need this inherited function, it's hard to find in spack.  I'll remove it.

File: scripts/spack/packages/serac/package.py [hunk 3]
```diff
@@ -164,327 +137,198 @@ class Serac(CMakePackage, CudaPackage):
...
-        cfg.write(cmake_cache_entry("AXOM_DIR", axom_dir))
-
-        conduit_dir = get_spec_path(spec, "conduit", path_replacements)
-        cfg.write(cmake_cache_entry("CONDUIT_DIR", conduit_dir))
+            entries.append(cmake_cache_path("TPL_ROOT", tpl_root))
 
-        hdf5_dir = get_spec_path(spec, "hdf5", path_replacements)
-        cfg.write(cmake_cache_entry("HDF5_DIR", hdf5_dir))
+        # required tpls
+        for dep in ('axom', 'conduit', 'mfem', 'hdf5',
```
Comment (joshessman-llnl) [inline scripts/spack/packages/serac/package.py:241]: :+1: Much cleaner!

**Review by joshessman-llnl — APPROVED** [review]

**Pull request merged**